### PR TITLE
feat(llmobs): add ml_app tag to LLMObs spans

### PR DESCRIPTION
**Pull request by Yun-Kim**
This PR adds a `DD_LLMOBS_APP_NAME` env var which configures the value of the `ml_app` tag to set on LLMObs span events.
Note that this env var is for private use only, so this doesn't need a release note.

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.
- [x] If change touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/llmobs/_llmobs.py [hunk 1]
```diff
@@ -382,10 +382,11 @@ def _llmobs_span_event(self, span: Span) -> Dict[str, Any]:
     @staticmethod
     def _llmobs_tags(span: Span) -> List[str]:
         tags = [
-            "version:%s" % (config.version or ""),
-            "env:%s" % (config.env or ""),
-            "service:%s" % (span.service or ""),
+            "version:{}".format(config.version or ""),
```
Comment (mabdinur) [inline ddtrace/llmobs/_llmobs.py:385]: But: We could use f strings here

File: tests/contrib/openai/utils.py [hunk 2]
```diff
@@ -36,6 +36,7 @@ def _expected_llmobs_tags(error=None):
         "env:",
         "service:",
         "source:integration",
+        "ml_app:",
```
Comment (mabdinur) [inline tests/contrib/openai/utils.py:39]: Should this have a default value? In the tracer we we set service names to  "unnamed_python_service"

File: ddtrace/settings/config.py [hunk 3]
```diff
@@ -552,6 +552,7 @@ def __init__(self):
 
         self._llmobs_enabled = asbool(os.getenv("DD_LLMOBS_ENABLED", False))
         self._llmobs_sample_rate = float(os.getenv("DD_LLMOBS_SAMPLE_RATE", 1.0))
+        self._llmobs_ml_app = os.getenv("DD_LLMOBS_ML_APP", "")
```
Comment (mabdinur) [inline ddtrace/settings/config.py:555]: If we don't expect customers to set this envar we should prefix it with an underscore
```suggestion
        self._llmobs_ml_app = os.getenv("_DD_LLMOBS_ML_APP", "")
```

**Review by mabdinur — APPROVED** [review]
Left some nuts. Feel free to resolve. Overall it looks great!

File: ddtrace/settings/config.py (hunk 3 above)
Comment (Yun-Kim) [reply ddtrace/settings/config.py]: Good point - I'll just remove this from the config for now, as users might use it but I don't want to visibly add this to our public API.

File: tests/contrib/openai/utils.py (hunk 2 above)
Comment (Yun-Kim) [reply tests/contrib/openai/utils.py:39]: Thanks for the catch!

File: ddtrace/llmobs/_llmobs.py (hunk 1 above)
Comment (Yun-Kim) [reply ddtrace/llmobs/_llmobs.py:385]: Will do these as a bigger cleanup effort in a future PR

**Review by alexbarksdale — APPROVED** [review]

**Pull request merged**